### PR TITLE
Concurrency issue at analyze

### PR DESCRIPTION
**Pull request by rscampos**
### 1. Explain what the PR does

fix: #3903
fix: #3905 

Instead of terminating the program upon reaching the EOF (after all events have been read), ensure the program concludes only after all events have been fully processed.

### CODE REVIEW

**Comment by geyslan** [comment]
Going to review this now. 👍🏼

**Comment by geyslan** [comment]
The issues are reproducible indeed.

**Review by geyslan — APPROVED** [review]
LGTM.

Tested with a huge json (100000000 entries / 109G) as input and analyze processed all events accordingly.

Signalling Tracee to finish during the analyze process worked smoothly as well.

**Review by AlonZivony — DISMISSED** [review]
Small thing to add on the same note.
Overall LGTM

File: cmd/tracee/cmd/analyze.go [hunk 1]
```diff
@@ -182,7 +190,7 @@ func produce(ctx context.Context, done chan interface{}, inputFile *os.File, eng
 			return
 		default:
 			if !scanner.Scan() { // if EOF or error close the done channel and return
-				close(done)
+				close(engineInput)
```
[the remark this answers is not in the document]
Comment (geyslan) [reply cmd/tracee/cmd/analyze.go:193]: @rscampos could you address this?

**Review by geyslan — APPROVED** [review]
LGTM

**Comment by geyslan** [comment]
Many thanks @rscampos for your help on this.

**Pull request merged**